source/extensions/tracers/opencensus/opencensus_tracer_impl.cc

```diff
@@ -251,6 +270,14 @@ Driver::Driver(const envoy::config::trace::v3::OpenCensusConfig& oc_config,
       auto channel =
           grpc::CreateChannel(oc_config.stackdriver_address(), grpc::InsecureChannelCredentials());
       opts.trace_service_stub = ::google::devtools::cloudtrace::v2::TraceService::NewStub(channel);
+    } else if (oc_config.has_stackdriver_call_credentials() &&
+               oc_config.stackdriver_call_credentials().has_sts_service()) {
+      // If call credential is provided and is STS service, create a stub with that credential.
+      auto call_creds = getStackdriverCallCredential(oc_config);
+      auto channel_creds =
+          grpc::CompositeChannelCredentials(grpc::GoogleDefaultCredentials(), call_creds);
+      auto channel = grpc::CreateChannel(kGoogleStackdriverTraceAddress, channel_creds);
+      opts.trace_service_stub = ::google::devtools::cloudtrace::v2::TraceService::NewStub(channel);
```

Comment:
I think we need a unit test that covers these lines. Somewhere in https://github.com/envoyproxy/envoy/blob/253bff42ee36da6b2190c50e1e9984454d453bb7/test/extensions/tracers/opencensus/config_test.cc. I don't think it's possible to test STS end-to-end, or how to introspect gRPC stubs, but at least we should verify that the credentials if config says so.

Reply:
Added a unit test to verify that a tracer could be constructed with sts configured.